### DISCUSSION TITLE
[archived] "Conditionals" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9364] [General]
# argo-workflows/walk-through/conditionals/

https://argoproj.github.io/argo-workflows/walk-through/conditionals/

**Comment by pangqiqiang** [comment]
I install argo server in `argo` namespace? I can only run this example workflow in `argo` namespace, if I run in `default` namespace, I get an error:
```
Error (exit code 1): Post "https://10.96.0.1:443/apis/argoproj.io/v1alpha1/namespaces/default/workflowtaskresults": dial tcp 10.96.0.1:443: connect: connection refused
```
is there any method that I can run it success in `default` namespace?

**Reply by agilgur5** [reply]
Unrelated to this doc ("Conditionals"), but it depends on your [installation](https://argo-workflows.readthedocs.io/en/latest/installation/) and configurations, particularly whether you have a [cluster install or namespaced install](https://argo-workflows.readthedocs.io/en/latest/installation/#installation-options). In a namespaced install, Workflows in a different namespace would not be watched by the Controller.
It seems like you have a cluster install since the Workflow ran and errored. This isn't an RBAC error, which would be a `Permission Denied` 403 error; you're getting a `connection refused`, which I imagine might be due to a `NetworkPolicy` or some other firewall you have in your cluster. That's out of scope of Argo, but all of Argo's components need network access to the k8s apiserver.

**Comment by jatinkumarg** [comment]
Does conditionals work with DAG? Its not working for me for some reason, getting this error:

```
Invalid 'when' expression '{{tasks.getLatestFileGcp.outputs.result}} == 0': Invalid token: '{{' (hint: try wrapping the affected expression in quotes ("))
```

**Reply by agilgur5** [reply]
Conditionals do work with DAGs (see also [the other comment](https://github.com/argoproj/argo-workflows/discussions/9364#discussioncomment-6641246) on this page).
The error looks to have recognized the conditional, but is saying the expression has a syntax error. Possibly the quoting was off? Not sure without seeing the full DAG Workflow

**Comment by pangqiqiang** [comment]
DAGs seems not accept the conditional



------------------&nbsp;Original message&nbsp;------------------
From: "Jatin Kumar"; 
Sendtime: Tuesday, Aug 30, 2022 4:21 AM
To: "argoproj/argo-workflows"; 
Cc: ***@***.***>; "Comment"; 
Subject: Re: [argoproj/argo-workflows] argo-workflows/walk-through/conditionals/ (Discussion #9364)





 
Does conditionals work with DAG? Its not working for me for some reason, getting this error:
 Invalid 'when' expression '{{tasks.getLatestFileGcp.outputs.result}} == 0': Invalid token: '{{' (hint: try wrapping the affected expression in quotes ("))

**Reply by agilgur5** [reply]
This should work for DAGs, `when` has been available for DAGs [since 2018](https://github.com/argoproj/argo-workflows/commit/11024318c0e2a9106f8a8b4a719daba12adf9f36) (and this piece of source code has [not changed](https://github.com/argoproj/argo-workflows/blame/cb1713d01542a7233d9bcb6646cc3c3409c5d870/workflow/controller/dag.go#L550) since then).

If you have a minimal reproduction of a DAG not working with `when` conditionals, please file an issue
